src/main/java/pw/itr0/kaba/encrypt/PBE.java

```diff
@@ -12,65 +13,201 @@
+
+    /**
+     * Returns default PBE encrypter.
+     *
+     * @return default PBE encrypter
+     */
     public static Encrypter getEncrypter() {
-        return new Encrypter(SALT, 50);
+        return getEncrypter(DEFAULT_SALT, DEFAULT_ITERATION_COUNT);
     }
 
+    /**
+     * Returns PBE encrypter whose salt is {@code salt} and iterationCount is {@code iterationCount}.
+     *
+     * @param salt           salt of PBE encryption
+     * @param iterationCount iteration count of PBE encryption
+     * @return PBE encrypter
+     */
+    public static Encrypter getEncrypter(byte[] salt, int iterationCount) {
+        byte[] copied = new byte[salt.length];
+        System.arraycopy(salt, 0, copied, 0, salt.length);
+        return new Encrypter(copied, iterationCount);
+    }
+
+    /**
+     * Returns default PBE decrypter.
+     *
+     * @return default PBE decrypter
+     */
     public static Decrypter getDecrypter() {
-        return new Decrypter(SALT, 50);
+        return getDecrypter(DEFAULT_SALT, DEFAULT_ITERATION_COUNT);
     }
 
-    private static Cipher getCipher(int mode, char[] password, byte[] salt, int iterationCount) throws NoSuchAlgorithmException,
-            InvalidKeySpecException, NoSuchPaddingException, InvalidAlgorithmParameterException, InvalidKeyException {
-        PBEKeySpec keySpec = new PBEKeySpec(password);
-        SecretKeyFactory keyFactory = SecretKeyFactory.getInstance(ENCRYPTION_ALGORITHM);
-        SecretKey key = keyFactory.generateSecret(keySpec);
+    /**
+     * Returns PBE decrypter whose salt is {@code salt} and iterationCount is {@code iterationCount}.
+     *
+     * @param salt           salt of PBE decryption
+     * @param iterationCount iteration count of PBE decryption
+     * @return PBE decrypter
+     */
+    public static Decrypter getDecrypter(byte[] salt, int iterationCount) {
+        byte[] copied = new byte[salt.length];
+        System.arraycopy(salt, 0, copied, 0, salt.length);
+        return new Decrypter(copied, iterationCount);
+    }
 
-        PBEParameterSpec paramSpec = new PBEParameterSpec(salt, iterationCount);
+    /**
+     * Returns {@link Cipher} for PBE encryption/decryption.
+     *
+     * @param mode           encryption({@link Cipher#ENCRYPT_MODE})/decryption({@link Cipher#DECRYPT_MODE})
+     * @param password       password of PBE
+     * @param salt           salt of PBE
+     * @param iterationCount iteration count of PBE
+     * @return {@link Cipher} for PBE encryption/decryption
+     * @throws NoSuchAlgorithmException if the running JRE does not support "{@value ENCRYPTION_ALGORITHM}"
+     * @throws InvalidKeyException      if the given key size exceeds the maximum allowable key size
+     */
+    private static Cipher getCipher(int mode, char[] password, byte[] salt, int iterationCount) throws NoSuchAlgorithmException, InvalidKeyException {
+
+        Cipher cipher;
+        try {
+            cipher = Cipher.getInstance(ENCRYPTION_ALGORITHM);
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Found 'DU'-anomaly for variable 'cipher' (lines '107'-'123').](https://www.codacy.com/app/ryotan/kaba-vault/file/2578626682/issues/source?bid=3298589&fileBranchId=3301306#l107)